src/status_im2/common/standard_auth/keycard/view.cljs

```diff
@@ -0,0 +1,178 @@
+(ns status-im2.common.standard-auth.keycard.view
+  (:require [react-native.core :as rn]
+            [quo2.core :as quo]
+            [status-im2.common.standard-auth.keycard.style :as style]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [status-im2.common.resources :as resources]
+            [utils.i18n :as i18n]
+            [quo2.components.keycard.view :as keycard]
+            [utils.re-frame :as rf]))
+
+(defn handle-number-entry
+  [entered-numbers max-digits max-attempt-reached number]
+  (when (>= (count @entered-numbers) max-digits)
+    (reset! entered-numbers []))
+  (when (= (count @entered-numbers) (dec max-digits))
+    (reset! max-attempt-reached true))
+  (swap! entered-numbers conj number))
+
+(defn enter-keycard-pin-sheet
+  []
+  (let [entered-numbers     (reagent/atom [])
+        max-digits          6
+        max-attempt-reached (reagent/atom false)]
+
+    (fn []
+      [rn/view {:style style/container}
+       [rn/view {:style style/inner-container}
+        [quo/text
+         {:accessibility-label :enter-keycard-pin-text
+          :size                :heading-2
+          :weight              :semi-bold}
+         (i18n/label :t/enter-keycard-pin)]
+
+        [rn/view {:style style/context-tag}
+         [quo/context-tag
+          {:type    :icon
+           :icon    :i/placeholder
+           :size    24
+           :context (i18n/label :t/card-with-name {:name "Alisher"})}]]
+
+        [rn/view (style/get-digits-container-style @max-attempt-reached)
+         (for [i (range max-digits)]
+           [rn/view
+            {:key   i
+             :style (style/digit-style @max-attempt-reached i entered-numbers)}])]
```

Comment:
If you have stateful code inside `for` you need to realize the lazy sequence, which is the case here. Either use `mapv` or the more common approach to surround `for` with `doall`.

Reply:
@ilmotta Makes sense. I updated my code with surrounding it with `doall`.